ci-operator/step-registry/idp/openldap/idp-openldap-commands.sh

```diff
@@ -0,0 +1,284 @@
+ou: people
+description: Users
+" > "$temp_ldif"
+
+    echo "Generating random passwords and updating temporary ldif file"
+    for i in $(seq 1 50); do
+        username="Person${i}"
+        password=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+        users+="${username}:${password},"
+
+        # Append user entry directly to the temporary ldif file
+        echo "
+dn: cn=$username,ou=people,ou=rfc2307,dc=example,dc=com
+objectClass: person
+objectClass: organizationalPerson
+objectClass: inetOrgPerson
+sn: Smith
+cn: $username
+displayName: ${username}smith
+mail: ${username}smith@example.com
+userPassword: $password
+" >> "$temp_ldif"
+    done
+
+    # Optionally, you can output the path of the temporary file
+    echo "Updated ldif file is located at: $temp_ldif"
+}
+
+
+function configure_openldap() {
+    echo "Configuring OpenLDAP in OpenShift"
+    oc create ns ldap
+
+    echo "Creating a secret from the updated temporary init.ldif file"
+    oc create secret generic ldap-init-ldif --from-file=init.ldif="$temp_ldif" -n ldap
+    oc adm policy add-scc-to-user anyuid -z default -n ldap
+
+    #Set ldap root password Secret
+    LDAP_SECRET=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+    oc create secret generic ldap-admin-password --from-literal=OPENLDAP_ROOT_PASSWORD="${LDAP_SECRET}" -n ldap
+
+    # Define the LDAP deployment
+    cat <<EOF | oc apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ldap
+  namespace: ldap
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: ldap
+  template:
+    metadata:
+      labels:
+        app: ldap
+    spec:
+      containers:
+      - name: ldap
+        image: quay.io/openshifttest/ldap:multiarch
+        env:
+        - name: OPENLDAP_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: ldap-admin-password
+              key: OPENLDAP_ROOT_PASSWORD
+        lifecycle:
+          postStart:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - |
+                sleep 120
+                # Then run ldapadd command
+                ldapadd -x -H 'ldap://127.0.0.1:389' -D 'cn=Manager,dc=example,dc=com' -w \$OPENLDAP_ROOT_PASSWORD -f /tmp/init.ldif
+        volumeMounts:
+        - name: init-ldif-volume
+          mountPath: /tmp
+      volumes:
+      - name: init-ldif-volume
+        secret:
+          secretName: ldap-init-ldif
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ldap
+  namespace: ldap
+spec:
+  selector:
+    app: ldap
+  ports:
+  - protocol: TCP
+    port: 389
+    targetPort: 389
+EOF
+
+    # Wait for LDAP pod to be running and ready
+    echo "Checking for LDAP pod to be running and ready"
+    count=0
+    max_attempts=10
+    while :; do
+        pod_status=$(oc get pods -n ldap --no-headers | grep ldap)
+        if [[ $pod_status == *"1/1"* && $pod_status == *"Running"* ]]; then
```

Comment:
Not sure if one single `[[ $pod_status == *1/1*Running* ]]` is good.

Reply:
Updated